### PR TITLE
Add forestopenfermion

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by BastianZim** [comment]
Waiting on https://github.com/rigetti/forest-openfermion/issues/21

**Comment by BastianZim** [comment]
@conda-forge/help-python This would be ready for review, thanks!

File: recipes/forestopenfermion/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - pip
+    - python >=3.6
+  run:
+    - numpy >=1.0.0,<2.0.0
+    - openfermion <1.0.0
+    - pyquil >=2.0.0,<3.0.0
+    - python >=3.6
+    - grove <2.0.0
+    - scipy >=1.0.0,<2.0.0
+    - mock
```
Comment (dopplershift) [inline recipes/forestopenfermion/meta.yaml:29]: What's driving the need for this one? I don't see it anywhere upstream.

Comment (BastianZim) [reply recipes/forestopenfermion/meta.yaml:29]: grove requires it and I'm fixing that on an interim basis until it's added upstream: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=420522&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1485

Comment (dopplershift) [reply recipes/forestopenfermion/meta.yaml:29]: Ah, ok.

**Review by dopplershift — APPROVED** [review]

**Comment by BastianZim** [comment]
Thanks for merging!

**Pull request merged**